### PR TITLE
MODGOBI-30 Default mappings 

### PR DESCRIPTION
**Pull request by javalkar-varun**
Jira - https://issues.folio.org/projects/MODGOBI/issues/MODGOBI-30

### CODE REVIEW

File: src/main/java/org/folio/gobi/MappingHelper.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import org.folio.gobi.Mapper.Translation;
+import org.folio.rest.impl.PostGobiOrdersHelper;
+import org.folio.rest.mappings.model.Mapping;
+import org.folio.rest.mappings.model.Mappings;
+import org.folio.rest.mappings.model.OrderMapping;
+import org.folio.rest.mappings.model.OrderMapping.OrderType;
+import org.w3c.dom.NodeList;
+import io.vertx.core.json.Json;
+
+public class MappingHelper {
```
[the remark this answers is not in the document]
Comment (javalkar-varun) [reply src/main/java/org/folio/gobi/MappingHelper.java]: Done

File: src/main/java/org/folio/rest/impl/InitConfigService.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+
+public class InitConfigService implements PostDeployVerticle {
+
+  @Override
+  public void init(Vertx vertx, Context context, Handler<AsyncResult<Boolean>> handler) {
+
+    System.out.println("Init Config Service");
```
[the remark this answers is not in the document]
Comment (javalkar-varun) [reply src/main/java/org/folio/rest/impl/InitConfigService.java]: Done. Using Logger.

File: src/main/java/org/folio/gobi/MappingHelper.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    try {
+      final InputStream is = PostGobiOrdersHelper.class.getClassLoader().getResourceAsStream(path);
+      if (is != null) {
+        return IOUtils.toString(is, "UTF-8");
+      } else {
+        return "";
+      }
+    } catch (IOException e) {
+      logger.error(String.format("Unable to read mock configuration in %s file", path));
+      e.printStackTrace();
```
[the remark this answers is not in the document]
Comment (mreno-EBSCO) [reply src/main/java/org/folio/gobi/MappingHelper.java]: I agree with @id-jenkins here. Don't do `e.printStackTrace()`. Use the logger: `logger.error(String.format("whatever"), e);`.

File: src/main/java/org/folio/gobi/Mapper.java [hunk 4]
```diff
@@ -215,7 +191,7 @@ public static String multiply(NodeList nodes) {
   }
 
   public static interface Translation<T> {
-    public CompletableFuture<T> apply(String s);
+    public CompletableFuture<T> apply(String s, PostGobiOrdersHelper gobiHelper);
```
Comment (craigmcnally) [inline src/main/java/org/folio/gobi/Mapper.java]: Why do we need to pass in the PostGobiOrdersHelper?

File: src/main/java/org/folio/rest/impl/PostGobiOrdersHelper.java [hunk 5]
```diff
@@ -64,98 +65,100 @@ public PostGobiOrdersHelper(HttpClientInterface httpClient,
...
+      mappings.put(Mapping.Field.MATERIAL_TYPE, DataSource.builder()
         .withFrom("//LocalData[Description='LocalData1']/Value")
-        .withTranslation(this::lookupMaterialTypeId)
+        //.withTranslation(this::lookupMaterialTypeId)
         .build());
-      mappings.put(Field.LOCATION, DataSource.builder()
+      mappings.put(Mapping.Field.LOCATION, DataSource.builder()
         .withFrom("//Location")
-        .withTranslation(this::lookupLocationId)
+        //.withTranslation(this::lookupLocationId)
```
Comment (craigmcnally) [inline src/main/java/org/folio/rest/impl/PostGobiOrdersHelper.java]: why are these commented out?

File: src/test/java/org/folio/gobi/MappingTest.java [hunk 6]
```diff
@@ -3,29 +3,43 @@
...
+  private PostGobiOrdersHelper postGobiOrdersHelper;
+  private Vertx vertx = Vertx.vertx();
+  private int port = NetworkUtils.nextFreePort();
+  private Map<String, String> okapiHeaders = new HashMap<>();
 
   @Before
   public void setUp() throws Exception {
     InputStream data = this.getClass().getClassLoader().getResourceAsStream(testdataPath);
     doc = DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(data);
+    okapiHeaders.put("X-Okapi-Url", "http://localhost:" + port);
```
Comment (craigmcnally) [inline src/test/java/org/folio/gobi/MappingTest.java]: please be consistent with case

File: src/main/java/org/folio/gobi/Mapper.java [hunk 7]
```diff
@@ -77,75 +53,75 @@ public Mapper(Map<Field, DataSource> mappings) {
 
       List<CompletableFuture<?>> futures = new ArrayList<>();
 
-      futures.add(mappings.get(Field.CREATED_BY)
-        .resolve(doc)
+      futures.add(mappings.get(Mapping.Field.CREATED_BY)
```
Comment (craigmcnally) [inline src/main/java/org/folio/gobi/Mapper.java]: I thought the point of this was to remove the hardcoded mappings in the code?

File: src/main/resources/data-mapping.json [hunk 8]
```diff
@@ -716,12 +762,95 @@
...
+            "translation": "getPurchaseOptionCode"
+          }
+        }          
       ]
     },
    {
       "orderType": "ListedPrintSerial",
       "mappings": [
+        {
+          "field": "MATERIAL_TYPE",
```
Comment (KVupp) [inline src/main/resources/data-mapping.json]: The local data fields will not be present in the listed print serial GOBI order xml. Same with all the below fields that are added additionally

File: src/main/java/org/folio/gobi/Mapper.java (hunk 7 above)
Comment (javalkar-varun) [reply src/main/java/org/folio/gobi/Mapper.java]: Hardcoded mappings removed.

**Comment by craigmcnally** [comment]
OK, got this to a state where it's building and passing PR checks...  Last piece is to address those TODO comments I made for myself

**Comment by craigmcnally** [comment]
FYI I removed my last comment (related to TODOs) since it wasn't intended for this PR!  Sorry for the confusion.

**Pull request merged**